src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/UnpooledDataSource.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET8_0_OR_GREATER
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX
```

Comment:
I tried the upgrade to 10 and didn't see any obvious compiler warnings or anything. So it may be straightforward? 🤞 
I'll do the change in a separate PR to keep it isolated.

Reply:
Yeah, some (most?) newer C# version features don't require targetting a new .NET TFM; for example, this feature (file-scoped namespace declarations) is purely syntactic, and can be used even if targetting netstandard2.0; so there's no reason to avoid using newer C# versions even when targetting old TFMs.